spring-cloud-azure-config/src/main/java/com/microsoft/azure/spring/cloud/config/AzureConfigBootstrapConfiguration.java

```diff
@@ -54,6 +60,24 @@ public ConnectionStringPool initConnectionString(AzureCloudConfigProperties prop
         return pool;
     }
 
+    @Bean
+    public AzureTokenCredentials tokenCredentials(AzureCloudConfigProperties properties) {
+        if (StringUtils.hasText(System.getenv(ENV_MSI_ENDPOINT))
+                && StringUtils.hasText(System.getenv(ENV_MSI_SECRET))) {
+            return new AppServiceMSICredentials(AzureEnvironment.AZURE);
+        }
+
+        AzureCloudConfigMSIProperties msiProps = properties.getMsi();
+        MSICredentials credentials = new MSICredentials();
+        if (msiProps != null && msiProps.getClientId() != null) {
+            credentials.withClientId(msiProps.getClientId());
+        } else if (msiProps != null && msiProps.getObjectId() != null) {
+            credentials.withObjectId(msiProps.getObjectId());
+        }
```

Comment:
this part shows that the clientId has high priority if both clientId and objectId exist, right ?